multinode/nodes/service.go

```diff
@@ -368,15 +332,50 @@ func (service *Service) trustedSatellites(ctx context.Context, node Node) (_ sto
 	return nodeURLs, nil
 }
 
-// nodeStatus chooses node status offline or online depends on LastContact.
-func nodeStatus(lastContact time.Time) Status {
-	now := time.Now().UTC()
 
-	if now.Sub(lastContact) < time.Hour*3 {
-		return StatusOnline
+// Fetch meta information about a node status, version, and last contact.
+func (service *Service) FetchNodeMeta(ctx context.Context, node Node) (_ Status,_*multinodepb.VersionResponse,_ *multinodepb.LastContactResponse) {
+	conn, err := service.dialer.DialNodeURL(ctx, storj.NodeURL{
+		ID:      node.ID,
+		Address: node.PublicAddress,
+	})
+	if err != nil {
+		return StatusNotReachable, nil, nil
 	}
 
-	return StatusOffline
+	defer func() {
+		err = errs.Combine(err, conn.Close())
+	}()
+
+	nodeClient := multinodepb.NewDRPCNodeClient(conn)
+
+	header := &multinodepb.RequestHeader{
+		ApiKey: node.APISecret[:],
+	}
+
+	nodeVersion, err := nodeClient.Version(ctx, &multinodepb.VersionRequest{Header: header})
+	if err != nil {
+		if rpcstatus.Code(err) == rpcstatus.Unauthenticated {
+			return StatusUnauthorized, nil,nil
+		}
+
+		return StatusStorageNodeInternalError, nil,nil
```

Comment:
Like I said above, we can still log the error and return the Internal server status